x-pack/metricbeat/module/gcp/metrics/compute/metadata.go

```diff
@@ -188,6 +189,12 @@ func (s *metadataCollector) getComputeInstances(ctx context.Context) {
 
 	defer instancesClient.Close()
 
+	start := time.Now()
+	defer func() {
+		totalTime := time.Since(start)
+		s.logger.Debugf("Total time taken for compute AggregatedList request: %s", totalTime)
+	}()
```

Comment:
We can reduce space complexity,

```suggestion
	defer func() {
		s.logger.Debugf("Total time taken for compute AggregatedList request: %s", time.Since(start))
	}()
```